### DISCUSSION TITLE
Suggestion: Basic React Packages

### DISCUSSION THREAD

**Discussion by FredMT** [wwwdevcc/projects#7] [Ideas]
# Kxlugu's Frontend Architecture Suggestions

### Routing
- **[TanStack Router](https://tanstack.com/router/latest/docs/framework/react/overview)** 
  - Type-safe routing with first-class search params
  - Automatic route pre-fetching
  - Built-in loader functions
  - Integrates with React Query for data fetching
  - Handles nested layouts elegantly

### Server State Management
- **[React Query](https://tanstack.com/query/latest/docs/framework/react/overview)**
  - We know what it does

### Styling
- **[Tailwind CSS](https://tailwindcss.com/docs/installation)**
  - Utility-first approach

### Component Library
- **[shadcn/ui](https://ui.shadcn.com/docs)**
  - Probably the best out there besides Mantine UI

### Form Management
- **[react-hook-form](https://react-hook-form.com/get-started)** (Alternative: [Tanstack Form](https://tanstack.com/form/latest/docs/overview))

**Comment by CydoEntis** [comment]
- Never used TanStack Router but I have heard all that good things about it so I am game to give it a go.
- Is react query enough for state management? I've always used it in conjunction with another state management library such as Zustand.
- Love me some tailwind
- Loving mantine UI atm using it in my personal project, but shadcn is also sick, if you use mantine ui you can use their built in form manager and not to use react-hook-form but im also familiar with react-hook-form and that was always my go to for form management

Should probably toss in Zod for some client side validation as well.

**Reply by FredMT** [reply]
> Is react query enough for state management? I've always used it in conjunction with another state management library such as Zustand.

I actually did include Zod and Zustand when I was writing down the md file. Removed it as I intended this to just be the base packages we start the project with. But those are definitely my recommendations as well for validation and client state management.

> Loving mantine UI atm using it in my personal project, but shadcn is also sick, if you use mantine ui you can use their built in form manager and not to use react-hook-form but im also familiar with react-hook-form and that was always my go to for form management

I've never actually tried Mantine UI, just seen the docs. And it seems pretty good. I've used shadcn/ui in literally every project I've done, and I love that it's so customizable. Would love to try out Mantine UI this time.

**Reply by CydoEntis** [reply]
Completely up to you, ive only dabbled in Shadcn we can use either or, I dont mind seeing how other people do things in other UI libraries. Mantine docs are definitely great but the library does have somethings I find pretty annoying, but I've been really enjoying myself after getting over that initial hurdle of figuring things out.

**Reply by FredMT** [reply]
> somethings I find pretty annoying

What exactly are the annoying parts?

**Reply by CydoEntis** [reply]
Theming doesn't feel the most intuitive with every component, and having to modify styling using the styles api with css modules as opposed to tailwindcss gets annoying, I also just have an issue with how the Paper component works with border and how there is nothing like borderColor prop to change border color. Sometimes the documentation doesn't feel clear enough on how to update a specific thing.

That said the pros of the library out weigh the cons, I've almost fully switched to using it for everything I work on now

**Reply by FredMT** [reply]
> That said the pros of the library out weigh the cons, I've almost fully switched to using it for everything I work on now

Mantine UI fully gets my vote then! I'm a little bit tired of making custom components all the time for common use-cases.
